### DISCUSSION TITLE
[Question][Hw3] How can I create an instrumented ll file?

### DISCUSSION THREAD

**Discussion by jiho-kwak** [prosyslab-classroom/cs348-information-security#515] [Q&A]
When I run _test/example1.instrumented.ll ./sanitizer test/example1.ll_ after make, it just gives me 'bash: test/example1.instrumented.ll: No such file or directory'.

So I've created an empty example1.instrumented.ll file, tried again, but it also didn't work.

How can create an instrumented ll file?

**Comment by Junseon00** [comment]
You have to run only './sanitizer test/example1.ll'
it will make the instrumented ll file on test directory

**Comment by jiho-kwak** [comment]
It seems like example1.ll is generated when I try make inside test directory, not root directory.
Also,  I've tried just './sanitizer test/example1.ll', but it also didn't created instrumented ll file.

**Comment by KihongHeo** [comment]
Can you take a look at `test/example1.instrumented.ll`?

**Comment by jiho-kwak** [comment]
test/example1.instrumented.ll isn't created at all

**Comment by KihongHeo** [comment]
Have you finished your implementation?
What is the output message of `./sanitizer test/example1.ll`?

**Comment by jiho-kwak** [comment]
I haven't finished implementation, I just wanted to check how the whole process works.
The output message is -  Fatal error: exception Failure("Not implemented")

**Comment by KihongHeo** [comment]
That is the reason.
The program terminates with "Not implemented" exception.
Please try to implement the thing first.

If you really want to see the whole process before starting homework, just try to make an "identity" sanitizer such as
```ocaml
let run llctx llm = llm
```

**Comment by jiho-kwak** [comment]
Thanks for your help!
